File: src/main/java/dev/zprestige/fire/module/movement/longjump/PacketReceiveListener.java
package dev.zprestige.fire.module.movement.longjump;

import dev.zprestige.fire.newbus.EventListener;
import dev.zprestige.fire.newbus.events.PacketEvent;
import net.minecraft.network.play.server.SPacketPlayerPosLook;

public class PacketReceiveListener extends EventListener<PacketEvent.PacketReceiveEvent, LongJump> {

    public PacketReceiveListener(final LongJump longJump){
        super(PacketEvent.PacketReceiveEvent.class, longJump);
    }

    @Override
    public void invoke(final Object object){
        final PacketEvent.PacketReceiveEvent event = (PacketEvent.PacketReceiveEvent) object;
        if (event.getPacket() instanceof SPacketPlayerPosLook) {
            final SPacketPlayerPosLook packet = (SPacketPlayerPosLook) event.getPacket();
            if (module.disableOnLag.GetSwitch()) {
                module.disableModule();
            }
            module.motionSpeed = 0.0f;
            mc.player.setPosition(packet.getX(), packet.getY(), packet.getZ());
        }
    }
}

File: src/main/java/dev/zprestige/fire/module/movement/longjump/PacketReceiveListener.kt
package dev.zprestige.fire.module.movement.longjump

import dev.zprestige.fire.event.bus.EventListener
import dev.zprestige.fire.event.impl.PacketEvent.PacketReceiveEvent
import net.minecraft.network.play.server.SPacketPlayerPosLook

class PacketReceiveListener(longJump: LongJump) : EventListener<PacketReceiveEvent, LongJump>(PacketReceiveEvent::class.java, longJump) {
    
    override fun invoke(e: Any) {
        val event = e as PacketReceiveEvent
        if (event.packet is SPacketPlayerPosLook) {
            val packet = event.packet as SPacketPlayerPosLook
            if (module.disableOnLag.GetSwitch()) {
                module.disableModule()
            }
            module.motionSpeed = 0.0
            mc.player.setPosition(packet.getX(), packet.getY(), packet.getZ())
        }
    }
}
